### DISCUSSION TITLE
CogAgent是否支持RAG？

### DISCUSSION THREAD

**Discussion by truebit** [THUDM/CogVLM#300] [Q&A] [answered]
CogAgent如何生成图片&文字的embeddings？
是否可以用RAG，提升grounding的结果呢？

**Comment by zRzRzRzRzRzRzR** [comment]
目前没有做相关工作，不过我认为多模态的RAG将会是未来大模型领域的一个研究方向

**Comment by truebit** [comment]
由于目前CogVLM只支持一张图片；如果用户自己使用文字RAG，是否可以改变visual grounding的结果呢？

**Reply by zRzRzRzRzRzRzR** [reply]
文字 RAG 肯定是可以的，但是改变visual grounding的结果是因为没有检测到正确的内容吗

**Reply by truebit** [reply]
是的，在领域场景中，CogAgent返回的Next Action是错误的。我在尝试使用RAG增加背景信息，使其更好的决策。
推理的时候传入history里的信息好像作用不大。是否可以理解为，只要每次传的是不一样的图片，我都可以认为是一个新的session？
然后在新session里面把所有的的之前的步骤信息和RAG内容都放到user query里面，发起推理呢？

**Reply by zRzRzRzRzRzRzR** [reply, marked as answer]
是的，每次传入图片就是一个新的session
但是，训练中的格式是
[图片]
对话一
对话二
。。。
这会导致其实他会吧所有内容以为都是对上传的图进行处理
